### PR TITLE
feat: New Module `avm-res-powerbidedicated-capacity`

### PR DESCRIPTION
**Pull request by ChrisSidebotham**
# Description

| Pipeline | 
| --------- | 
| [avm.res.power-bi-dedicated.capacity](https://github.com/ChrisSidebotham/bicep-registry-modules/actions/workflows/avm.res.power-bi-dedicated.capacity.yml) |

### CODE REVIEW

File: .github/workflows/avm.res.power-bi-dedicated.capacity.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
+name: "avm.res.power-bi-dedicated.capacity"
```
Comment (AlexanderSehr) [inline .github/workflows/avm.res.power-bi-dedicated.capacity.yml:1]: Just a small blocking comment to give this PR a break until the update registry names are available.
No todo for you @ChrisSidebotham . I'll review the rest of the PR soon

Comment (ChrisSidebotham) [reply .github/workflows/avm.res.power-bi-dedicated.capacity.yml:1]: Hoping to have nailed this one 🤌

File: avm/res/power-bi-dedicated/capacity/main.bicep [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+  'Log Analytics Contributor': subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '92aaf0da-9dab-42b6-94a3-d43ce8d16293')
+  'Log Analytics Reader': subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '73c42c96-874c-492b-b04d-ab87d138a893')
+  Owner: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '8e3af657-a8ff-443c-a75c-2fe8c4bcb635')
+  Reader: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'acdd72a7-3385-48ef-bd42-f606fba81ae7')
+  'Role Based Access Control Administrator (Preview)': subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'f58310d9-a9f6-439a-9e8d-f62e7b41a168')
+  'User Access Administrator': subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '18d7d88d-d35e-4fb5-a5c3-7773c20a72d9')
+}
+
+resource defaultTelemetry 'Microsoft.Resources/deployments@2021-04-01' = if (enableTelemetry) {
+  name: '46d3xbcp.res.batch-batchaccount.${replace('-..--..-', '.', '-')}.${substring(uniqueString(deployment().name, location), 0, 4)}'
```
Comment (AlexanderSehr) [inline avm/res/power-bi-dedicated/capacity/main.bicep:47]: Incorrect telemetry

File: avm/res/power-bi-dedicated/capacity/main.bicep [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+  @description('Optional. Version of the condition.')
+  conditionVersion: '2.0'?
+
+  @description('Optional. The Resource Id of the delegated managed identity resource.')
+  delegatedManagedIdentityResourceId: string?
+}[]?
+
+type skuTpe = {
+  @description('Optional. The name of the SKU.')
+  name: ('A1' | 'A2' | 'A3' | 'A4' | 'A5' | 'A6' | null)?
```
Comment (AlexanderSehr) [inline avm/res/power-bi-dedicated/capacity/main.bicep:153]:
```suggestion
  name: ('A1' | 'A2' | 'A3' | 'A4' | 'A5' | 'A6')?
```
Null should be a given via the `?`

File: avm/res/power-bi-dedicated/capacity/main.bicep [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+
+  @description('Required. The capacity of the SKU.')
+  capacity: int
+
+  @description('Optional. The tier of the SKU.')
+  tier: ('PBIE_Azure' | 'Premium' | 'AutoPremiumHost' | null)?
```
Comment (AlexanderSehr) [inline avm/res/power-bi-dedicated/capacity/main.bicep:159]:
```suggestion
  tier: ('PBIE_Azure' | 'Premium' | 'AutoPremiumHost')?
```
Null should be a given via the `?`

File: .github/workflows/avm.res.power-bi-dedicated.capacity.yml (hunk 1 above)
Comment (AlexanderSehr) [reply .github/workflows/avm.res.power-bi-dedicated.capacity.yml:1]: @ChrisSidebotham please fetch from upstream and regenerate the readme :)

File: avm/res/power-bi-dedicated/capacity/main.bicep [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+
+type roleAssignmentType = {
+  @description('Required. The name of the role to assign. If it cannot be found you can specify the role definition ID instead.')
+  roleDefinitionIdOrName: string
+
+  @description('Required. The principal ID of the principal (user/group/identity) to assign the role to.')
+  principalId: string
+
+  @description('Optional. The principal type of the assigned principal ID.')
+  principalType: ('ServicePrincipal' | 'Group' | 'User' | 'ForeignGroup' | 'Device' | null)?
```
Comment (AlexanderSehr) [inline avm/res/power-bi-dedicated/capacity/main.bicep:136]:
```suggestion
  principalType: ('ServicePrincipal' | 'Group' | 'User' | 'ForeignGroup' | 'Device')?
```

Comment (AlexanderSehr) [reply avm/res/power-bi-dedicated/capacity/main.bicep:136]: Fixed recently in UDT spec

File: .github/workflows/avm.res.power-bi-dedicated.capacity.yml (hunk 1 above)
Comment (ChrisSidebotham) [reply .github/workflows/avm.res.power-bi-dedicated.capacity.yml:1]: Updated

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**